eth/ethconfig/estimate/esitmated_ram.go

```diff
@@ -4,33 +4,44 @@ import (
 	"runtime"
 
 	"github.com/c2h5oh/datasize"
+
 	"github.com/ledgerwatch/erigon-lib/common/cmp"
 	"github.com/ledgerwatch/erigon-lib/mmap"
 )
 
-type estimatedRamPerWorker datasize.ByteSize
+type EstimatedRamPerWorker datasize.ByteSize
 
 // Workers - return max workers amount based on total Memory/CPU's and estimated RAM per worker
-func (r estimatedRamPerWorker) Workers() int {
-	// 50% of TotalMemory. Better don't count on 100% because OOM Killer may have aggressive defaults and other software may need RAM
-	maxWorkersForGivenMemory := (mmap.TotalMemory() / 2) / uint64(r)
-	res := cmp.Min(AlmostAllCPUs(), int(maxWorkersForGivenMemory))
+func (r EstimatedRamPerWorker) Workers() int {
+	maxWorkersForGivenMemory := r.WorkersIgnoringCPUs()
+	res := cmp.Min(AlmostAllCPUs(), maxWorkersForGivenMemory)
 	return cmp.Max(1, res) // must have at-least 1 worker
 }
 
-func (r estimatedRamPerWorker) WorkersHalf() int    { return cmp.Max(1, r.Workers()/2) }
-func (r estimatedRamPerWorker) WorkersQuarter() int { return cmp.Max(1, r.Workers()/4) }
+func (r EstimatedRamPerWorker) WorkersHalf() int {
+	return cmp.Max(1, r.Workers()/2)
+}
+
+func (r EstimatedRamPerWorker) WorkersQuarter() int {
+	return cmp.Max(1, r.Workers()/4)
+}
+
+// WorkersIgnoringCPUs - return max workers amount based on total Memory and estimated RAM per worker
+func (r EstimatedRamPerWorker) WorkersIgnoringCPUs() int {
```

Comment:
suggest to rename as "WorkersByRAMOnly" or "WorkersForIO"

Reply:
done 👍